db/corruption_test.cc

```diff
@@ -1115,7 +1112,7 @@ TEST_P(CrashDuringRecoveryWithCorruptionTest, CrashDuringRecovery) {
     std::vector<uint64_t> file_nums;
     GetSortedWalFiles(file_nums);
     size_t size = file_nums.size();
-    uint64_t log_num = file_nums[size - 2];
+    uint64_t log_num = file_nums[size - 1];
```

Comment:
I tried corrupting second last WAL but I get the error "SST file ahead of WAL for default" because of sequence number. I am having difficulty understanding how to corrupt the second WAL and if this is bug or expected behavior.

Reply:
It seems expected to me. The second last WAL should be synced during `Flush()`. The last WAL might not be synced. But I don't see any error repro'ing when the last WAL is corrupted.

Reply:
I changed the TEST_SwitchMemtable to Flush and the started giving error without this change. Let me add a screenshot of error(without this change) in the Test plan which I ran on main branch.

Reply:
I think the test might really need to fail Open() after finishing manifest change for a subset of CFs flushed. After that the next Open() should fail even without any failure injection, assuming the current ordering of operations is problematic.

A subset of CFs can be flushed during `Open()` even when `avoid_flush_during_recovery == true` using `NewSpecialSkipListFactory(1)`. The failure could be injected for example into the "CURRENT" file `Sync()` to make sure the MANIFEST update is fully finished.

Reply:
Just to clarify, the current unit tests will fail without the fix, right?

Reply:
I also wonder if it's possible to reproduce the failure with a column family whose memtable is empty but has one L0. It seems that the log number of this column family will be increased even if no flush occurs during recovery.

Reply:
> Just to clarify, the current unit tests will fail without the fix, right?
Yes.

Reply:
> I also wonder if it's possible to reproduce the failure with a column family whose memtable is empty but has one L0. It seems that the log number of this column family will be increased even if no flush occurs during recovery.

Let me try to reproduce it.

Reply:
> > Just to clarify, the current unit tests will fail without the fix, right?
> > Yes.

Can you provide the code where it fails? Here is my attempt to run current corruption_test without the fix, where it passes -- https://github.com/ajkr/rocksdb/tree/pr9922 , 

```
$ make -j24 corruption_test && ./corruption_test
...
[----------] Global test environment tear-down
[==========] 33 tests from 2 test cases ran. (19067 ms total)
[  PASSED  ] 33 tests.
```

Reply:
Let me check. It shouldn't happen. It should fail without fix.

Reply:
@ajkr  It wasn't doing the full truncation (bug in corruption helper). I updated the test and now its failing with both the fix and without the fix. I need to look into it to understand why with the fix its failing.

I will try to maintain two branches and update both (with fix and without fix). 
Other branch is https://github.com/facebook/rocksdb/pull/9942.